.github/workflows/infrastructure-cleardown.yaml

```diff
@@ -0,0 +1,95 @@
+name: Cleardown Infrastructure
+
+on:
+  workflow_call:
+    inputs:
+      github_environment:
+        description: "Defines the Github environment in which to pull environment variables from"
+        required: true
+        type: string
+      branch_name:
+        description: "Name of the branch used to derive tf version and workspace"
+        required: true
+        default: 'not-set'
+        type: string
+
+jobs:
+  minMax:
+    name: "Derive TF Version"
```

Comment:
this is done as part of action-infrastructure-stack via use of deply-infrastructure below. But may need to be tweaked to handle reference to deleted branch